ios/FluentUI/Bottom Commanding/BottomCommandingController.swift

```diff
@@ -677,6 +697,49 @@ open class BottomCommandingController: UIViewController {
         return requiredConstraints + [breakableTopConstraint]
     }
 
+    // MARK: - Bottom bar animation
+
+    private func bottomBarHidingAnimator(to isHidden: Bool) -> UIViewPropertyAnimator? {
+        guard let bottomBarView = bottomBarView, let bottomConstraint = bottomBarViewBottomConstraint else {
+            return nil
+        }
+
+        let springParams = UISpringTimingParameters(dampingRatio: Constants.BottomBar.hidingSpringDamping)
+        let animator = UIViewPropertyAnimator(duration: Constants.BottomBar.hidingSpringDuration, timingParameters: springParams)
+
+        let initialBottomConstant = bottomConstraint.constant
+        let initialHiddenState = bottomBarView.isHidden
+
+        bottomConstraint.constant = isHidden ? -Constants.BottomBar.hiddenBottomOffset : -Constants.BottomBar.bottomOffset
+        bottomBarView.isHidden = false
+
+        animator.addAnimations { [weak self] in
+            self?.view.layoutIfNeeded()
+        }
+
+        animator.addCompletion { [weak self] finalPosition in
+            if finalPosition == .end {
+                bottomBarView.isHidden = isHidden
+                self?._isHidden = isHidden
```

Comment:
Might be safer to capture the strong self reference here to update the property.

Reply:
Why is it safer? For a single access to self it just seems redundant to me. Either way, if self is still alive - change the state. If it's not - don't. The logic would be the same right?